localstack-core/localstack/testing/pytest/fixtures.py

```diff
@@ -904,10 +904,10 @@ def opensearch_wait_for_cluster(aws_client):
     def _wait_for_cluster(domain_name: str):
         def finished_processing():
             status = aws_client.opensearch.describe_domain(DomainName=domain_name)["DomainStatus"]
-            return status["Processing"] is False
+            return status["Processing"] is False and "Endpoint" in status
```

Comment:
Why is the second condition necessary here?

Reply:
When opensearch domain is done processing there's a small gap where processing is false, but the endpoint is not ready yet.